control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go

```diff
@@ -247,7 +248,9 @@ func (r *Reconciler) reconcileInPlaceUpgrade(ctx context.Context, nodePoolUpgrad
 
 	// Find nodes that can be upgraded
 	// TODO (jerzhang): add logic to honor maxUnavailable/maxSurge
-	nodesToUpgrade := getNodesToUpgrade(nodes, targetConfigVersionHash, 1)
+	// Fetch the nodePool instance
+	nodePool := &hyperv1.NodePool{}
+	nodesToUpgrade := getNodesToUpgrade(nodes, targetConfigVersionHash, nodePool.Spec.Management.InPlace.MaxUnavailable.IntValue())
```

Comment:
@yuqi-zhang Should I check whether `MaxUnavailable` is a percentage instead of a scalar value and add logic to convert the percentage to a scalar value if it is?

Reply:
Yes, since in the above typing, you have it set to `*intstr.IntOrString`

I think a replication of the MCO logic https://github.com/openshift/machine-config-operator/blob/f947dbdab6c1380ce5ee4743b5780329ced693d5/pkg/controller/node/node_controller.go#L1159-L1172 should suffice for this